langkit/envs.py

```diff
@@ -235,25 +226,46 @@ def _parse_actions(self, actions: List[EnvAction]) -> None:
         Analyze the given list of actions and extract pre/post actions, i.e.
         actions executed before and after handling children.
         """
+        def count(cls: Type[EnvAction],
+                  sequence: Optional[List[EnvAction]] = None) -> int:
+            """
+            Return the number of ``cls`` instances in ``sequence``, or in
+            ``actions`` if ``sequence`` is left to None.
+            """
+            if sequence is None:
```

Comment:
I find this added logic a bit superfluous. Simply calling `count(..., actions)` below is way shorter and even makes it easier to understand what's going on.

Reply:
Good idea, thanks!